src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementSecureDocStoreService.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.reform.sscs.service;
+
+import java.net.URI;
+import java.util.List;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.document.am.feign.CaseDocumentClientApi;
+import uk.gov.hmcts.reform.ccd.document.am.model.Classification;
+import uk.gov.hmcts.reform.ccd.document.am.model.Document;
+import uk.gov.hmcts.reform.ccd.document.am.model.DocumentUploadRequest;
+import uk.gov.hmcts.reform.ccd.document.am.model.UploadResponse;
+import uk.gov.hmcts.reform.sscs.document.EvidenceDownloadClientApi;
+import uk.gov.hmcts.reform.sscs.exception.UnsupportedDocumentTypeException;
+import uk.gov.hmcts.reform.sscs.idam.IdamTokens;
+
+@Service
+@Slf4j
+public class EvidenceManagementSecureDocStoreService {
+
+    public static final String S2S_TOKEN = "oauth2Token";
+
+    private final AuthTokenGenerator authTokenGenerator;
+    private final CaseDocumentClientApi caseDocumentClientApi;
+    private final EvidenceDownloadClientApi evidenceDownloadClientApi;
+
+    @Autowired
+    public EvidenceManagementSecureDocStoreService(
+        AuthTokenGenerator authTokenGenerator,
+        CaseDocumentClientApi caseDocumentClientApi,
+        EvidenceDownloadClientApi evidenceDownloadClientApi
+    ) {
+        this.authTokenGenerator = authTokenGenerator;
+        this.caseDocumentClientApi = caseDocumentClientApi;
+        this.evidenceDownloadClientApi = evidenceDownloadClientApi;
+    }
+
+    public UploadResponse upload(List<MultipartFile> files, IdamTokens idamTokens) {
+
+        String serviceAuthorization = authTokenGenerator.generate();
```

Comment:
Do we need this, can't we get that from the idamTokens?

Reply:
yes, your right